tools/bl.c

```diff
@@ -679,6 +679,38 @@ static int bl_setup_signal_handler(void)
 	return EXIT_SUCCESS;
 }
 
+static void bl_auto_dev(char *argv[])
+{
+	enum
+	{
+		ARG_PROG,
+		ARG_CMD,
+		ARG_DEV_PATH,
+		ARG__COUNT,
+	};
+
+	if ((strncmp(argv[ARG_DEV_PATH], "auto", 4) ||
```

Comment:
I'd pass in `argc`, and only do the rest if: `argc > ARG_DEV_PATH`.

Reply:
Ok, I'll do the check in main, as I think the function itself shouldn't be depends on the input argument format